### ISSUE TITLE
[a11y][iOS] Overlaid widgets are not accessible by voice controls on iOS

### ISSUE THREAD

**Issue by ChristianEdwardPadilla** [flutter/flutter#133885]
Internal: b/297151696

### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. On iOS, enable voice controls (Settings  > Accessibility > Voice Control, enable Show Hints)
2. Run the code sample (code below).
3. Observe that there is only one hint (the numeral 2) over the overlaid widgets (see screenshot for comparison with Android behavior).

### Expected results

(It is not known if this is intended behavior on iOS)
There should be some way to interact with overlaid widgets.

### Actual results

There is no way to interact with the overlaid widget via voice commands on iOS.

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('a11y overlay bug demo'),
      ),
      body: Stack(
        children: [
          ListView(
            shrinkWrap: true,
            padding: EdgeInsets.zero,
            children: <Widget>[
              Semantics(
                onTap: () {},
                child: Container(
                  height: 500,
                  color: Colors.amber[600],
                  child: const Center(child: Text('Entry A')),
                ),
              ),
              Container(
                height: 500,
                color: Colors.amber[500],
                child: const Center(child: Text('Entry B')),
              ),
              Container(
                height: 500,
                color: Colors.amber[100],
                child: const Center(child: Text('Entry C')),
              ),
            ],
          ),
          Semantics(
            // "Tap 1" voice command on Android triggers this onTap.
            // on iOS, without the padding line below, there is no way to tap this widget via voice controls.
            onTap: () => print('Toolbar container tapped.'),
            child: Container(
              height: 300,
              // Uncommenting this line will give the underlying widget enough space to be seen by the voice controls on iOS.
              // padding: const EdgeInsets.all(20.0),
              color: Colors.green,
              child: TextButton(
                // "Tap 2" voice command on Android triggers this onTap.
                onPressed: () => print('Toolbar container button tapped.'),
                child: const Text('Toolbar container below app bar'),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots of Android and iOS behavior</summary>
Android:

![Screenshot_20230824-152820](https://github.com/flutter/flutter/assets/37954976/e17e49a5-7d8a-4771-939d-51b590980861)

iOS:

![4ZVmjdgru6KuzXS](https://github.com/flutter/flutter/assets/37954976/79b4cea1-229e-4a04-96c3-dc4bed4dd060)
</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel google3, on macOS 13.5 22G74 darwin-x64, locale en)
    • Framework revision 579991cbb3 (7 days ago), 2023-08-17T00:00:00.000
    • Engine revision 5019d6da65
    • Dart version 7e4e5796ee

[✓] Android toolchain - develop for Android devices (Android SDK version Stable)
    • Android SDK at google3
    • Platform Stable, build-tools Stable
    • Java binary at: /Applications/Android Studio with
      Blaze.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] GoogleMac (on macos)
    • AppleWWDRCA installed.
    • Google development certificate installed.

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio with Blaze
      Beta.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b829.5-9884540)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c

[✓] Connected device (2 available)
    • Pixel 4 (mobile) • 99211FFAZ0054E • android-arm64 • Android 13 (API 33)
    • macOS (desktop)  • macos          • darwin-x64    • macOS 13.5 22G74
      darwin-x64

! Doctor found issues in 1 category.
```

</details>

**Comment by ChristianEdwardPadilla** [comment]
Internal tracking bug: b/297151696.

**Comment by jiahaog** [comment]
Updating the priority to match the internal bug

**Comment by stuartmorgan** [comment]
I did some investigation here, and I'm having trouble understanding what exactly the bug is based on the description and the internal discussion. In particular, the screenshots here don't match the sample code, and in my tests, this comment:
>  // Uncommenting this line will give the underlying widget enough space to be seen by the voice controls on iOS.
>  // padding: const EdgeInsets.all(20.0),

isn't correct for the sample code here, in that I don't see any change when adding this padding. It doesn't seem like the generic semantic nodes are visible no matter what in this example, even when completely removing layers, so it's not clear to me that the repro case here is actually correct.

Can we get a clearer, consistent set of steps and outcomes?

**Comment by stuartmorgan** [comment]
(As a general note about the internal discussion: iOS accessibility generally operates on a hit-test basis, so if it's impossible to tap a widget manually, I would not expect it to be surfaced for voice control on iOS.)

**Comment by ChristianEdwardPadilla** [comment]
```
(As a general note about the internal discussion: iOS accessibility generally operates on a hit-test basis, so if it's impossible to tap a widget manually, I would not expect it to be surfaced for voice control on iOS.)
```

This very well might be WAI then.

I'm not sure why you're not seeing differences in before/after commenting that line. Here is what I see on iOS without padding (first image) and with padding (second image):

![after](https://github.com/flutter/flutter/assets/37954976/a25a14e9-a390-421b-a4a7-0b48c61eaba6)

![before](https://github.com/flutter/flutter/assets/37954976/6244b919-594c-4185-829b-b290bb9e4dd6)
